client/workflows/client.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2021 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package workflows
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mendersoftware/go-lib-micro/identity"
+	"github.com/mendersoftware/go-lib-micro/requestid"
+	"github.com/mendersoftware/go-lib-micro/rest_utils"
+	"github.com/mendersoftware/inventory/utils"
+	"github.com/pkg/errors"
+)
+
+const (
+	ReindexURI = "/api/v1/workflow/reindex_reporting"
+	HealthURI  = "/api/v1/health"
+)
+
+const (
+	defaultTimeout = time.Duration(5) * time.Second
+)
+
+// Client is the workflows client
+//go:generate ../../utils/mockgen.sh
+type Client interface {
+	CheckHealth(ctx context.Context) error
+	StartReindex(c context.Context, device string) error
+}
+
+type ClientOptions struct {
+	Client *http.Client
+}
+
+// NewClient returns a new workflows client
+func NewClient(url string, opts ...ClientOptions) Client {
+	// Initialize default options
+	var clientOpts = ClientOptions{
+		Client: &http.Client{},
+	}
+	// Merge options
+	for _, opt := range opts {
+		if opt.Client != nil {
+			clientOpts.Client = opt.Client
+		}
+	}
+
+	return &client{
+		url:    strings.TrimSuffix(url, "/"),
+		client: *clientOpts.Client,
+	}
+}
+
+type client struct {
+	url    string
+	client http.Client
+}
+
+func (c *client) StartReindex(ctx context.Context, device string) error {
+	if _, ok := ctx.Deadline(); !ok {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, defaultTimeout)
+		defer cancel()
+	}
+	tenantID := ""
+	if id := identity.FromContext(ctx); id != nil {
+		tenantID = id.Tenant
+	}
+	wflow := ReindexWorkflow{
+		RequestID: requestid.FromContext(ctx),
+		TenantID:  tenantID,
+		DeviceID:  device,
+		Service:   ServiceInventory,
+	}
+	payload, _ := json.Marshal(wflow)
+	req, err := http.NewRequestWithContext(ctx,
+		"POST",
+		c.url+ReindexURI,
+		bytes.NewReader(payload),
+	)
+	if err != nil {
+		return errors.Wrap(err, "workflows: error preparing HTTP request")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	rsp, err := c.client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "workflows: failed to submit auditlog")
+	}
```

Comment:
*lint* Response body not closed
```suggestion
	}
	defer rsp.Body.Close()
```